### PR TITLE
Fixes and feature additions.

### PR DESCRIPTION
**Pull request by ejez**
Fix the uninstall script.
Make the extension functions async, see #82 and #81 
Add a `cfg` parameter to apollo creation functions, see #85 and #84

### CODE REVIEW

File: src/uninstall.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
 const fs = require('fs');
 
+module.exports = function (api) {
 // get the graphql html that will be removed from 'src/index.template.html'
-const graphqlHtml = require('./exports').graphqlHtml;
+  let graphqlHtml
+
+  if (api.getPackageVersion('@quasar/app') >= '2.0.0') {
```
Comment (hawkeye64) [inline src/uninstall.js:7]: This is a possible point of failure. Instead, use the `semver` package. Then you can do something like this:
```js
if (semver.gte(api.getPackageVersion('@quasar/app'), '2.0.0')) {
```

Comment (ejez) [reply src/uninstall.js]: Thanks @hawkeye64 

I checked the [Uninstall API](https://quasar.dev/app-extensions/development-guide/uninstall-api) and it does not have the function `getPackageVersion`, so for the moment I added a commit to revert the changes to `uninstall.js`.

We will need a way to get the package version of `@quasar/app`, or drop support for versions less than `2.0.0`

Comment (smolinari) [reply src/uninstall.js]: @ejez - Can you make that change please?

Scott

Comment (hawkeye64) [reply src/uninstall.js]: I have done the same as you guys are trying with both qenv and dotenv app-extensions

```js
  const version = api.getPackageVersion('@quasar/app')
  const v1 = semver.lt(version, '2.0.0')
```

If you are missing the api, then you need to do something like this:
```js
module.exports = function (api) {
  // Quasar compatibility check; you may need
  // hard dependencies, as in a minimum version of the "quasar"
  // package or a minimum version of "@quasar/app" CLI
  // api.compatibleWith('quasar', '^1.1.1')
  api.compatibleWith('@quasar/app', '^1.1.0 || ^2.0.0 || ^3.0.0-beta.1')

  // We extend /quasar.conf.js
  api.extendQuasarConf((conf) => {
    extendConf(api, conf)
  })
}
```

So you have api in your other hooks

Comment (hawkeye64) [reply src/uninstall.js]: a bit more code on that usage:
```js
  // previous tp @quasar/app v2.0.0 the data had to be stringified
  for (const key in parsed) {
    target[key] = v1 === true ? JSON.stringify(parsed[key]) : parsed[key]
  }
```

Comment (hawkeye64) [reply src/uninstall.js]: If the functions to the API are not the same as the Install api, we should create an issue ticket

Comment (hawkeye64) [reply src/uninstall.js]: I made the issue

**Comment by ejez** [comment]
I added a commit to fix the uninstall script.

While waiting for a fix to https://github.com/quasarframework/quasar/issues/8473 I borrowed the `InstallAPI` :)

**Review by smolinari — APPROVED** [review]

**Comment by hawkeye64** [comment]
Just so you know, the Uninstall now has `getPackageVersion` and `hasPackage` (if needed)

https://github.com/quasarframework/quasar/issues/8473

I would recommend making the adjustments instead of your current code.

**Comment by ejez** [comment]
@hawkeye64 
> Just so you know, the Uninstall now has getPackageVersion and hasPackage methods exposed.

Thanks!

> I would recommend making the adjustments instead of your current code.

`getPackageVersion` and `hasPackage` were added in branches `qv2-search` and `vue3-work` (https://github.com/quasarframework/quasar/commit/4416e770ef6ede84d81ea823233fdda5933071b0), they are not available in @quasar/app v1 or v2.

Even if they were added to v2, we still cannot use them unless we intend to drop support for previous v2 versions. The extension user might have a version of @quasar/app that does not have the added functions, and our extension code would break if it relies on them.

(interestingly, the code in this extension checks for the version of `@quasar/app` **_itself_**)

We can modify the code in the next major version of the extension to use the added functions.

**Comment by hawkeye64** [comment]
@ejez I should have looked at it a bit better. Since it will be available when "Quasar v2" / "@quasar/app v3" are released, you will have to test for it being there. I can see why it's not in earlier versions, but would have been nice.

**Pull request merged**